producer/cloudwatch_logs.go

```diff
@@ -0,0 +1,164 @@
+	describeLogstreamsDelay = 200 * time.Millisecond
+	// PutLogEvents 5 requests/second/log stream.
+	putLogEventsDelay = 200 * time.Millisecond
+)
+
+type CloudwatchLogs struct {
+	core.BufferedProducer `gollumdoc:"embed_type"`
+	stream                *string
+	group                 *string
+	token                 *string
+	service               *cloudwatchlogs.CloudWatchLogs
+}
+
+func init() {
+	core.TypeRegistry.Register(CloudwatchLogs{})
+}
+
+// Configure initializes this producer with values from a plugin config.
+func (prod *CloudwatchLogs) Configure(conf core.PluginConfigReader) error {
+	prod.SetStopCallback(prod.flush)
+	stream := conf.GetString("stream", "")
+	if stream == "" {
+		return fmt.Errorf("stream name can not be empty")
+	}
+	group := conf.GetString("group", "")
+	if group == "" {
+		return fmt.Errorf("group name can not be empty")
+	}
+	prod.stream = &stream
+	prog.group = &group
+	return nil
+}
+
+// Put log events and update sequence token.
+// Possible errors http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html
+func (prod *CloudwatchLogs) upload() {
+	logevents := make([]*cloudwatchlogs.InputLogEvent, 0, len(events))
+	for _, elem := range events {
+		logevents = append(logevents, &cloudwatchlogs.InputLogEvent{
+			Message:   aws.String(elem.msg),
+			Timestamp: aws.Int64(elem.timestamp),
+		})
+	}
+	params := &cloudwatchlogs.PutLogEventsInput{
+		LogEvents:     logevents,
+		LogGroupName:  prod.group,
+		LogStreamName: prod.stream,
+		SequenceToken: prod.token,
+	}
+	// When rejectedLogEventsInfo is not empty, app can not
+	// do anything reasonable with rejected logs. Ignore it.
+	// Meybe expose some statistics for rejected counters.
+	resp, err := dst.svc.PutLogEvents(prod.cloudwatchLogsParams)
+	if err == nil {
+		prod.cloudwatchLogsParams.SequenceToken = resp.NextSequenceToken
+	}
+}
+
+func (prod *Console) Produce(workers *sync.WaitGroup) {
+	defer prod.WorkerDone()
+	prod.AddMainWorker(workers)
+	prod.MessageControlLoop(prod.printMessage)
+}
+
+// For newly created log streams, token is an empty string.
+func (prod *CloudwatchLogs) setToken() error {
+	params := &cloudwatchlogs.DescribeLogStreamsInput{
+		LogGroupName:        prod.group,
+		LogStreamNamePrefix: prod.stream,
+	}
+
+	return prod.service.DescribeLogStreamsPages(params,
+		func(page *cloudwatchlogs.DescribeLogStreamsOutput, lastPage bool) bool {
+			return !findToken(prod, page)
+		})
+}
+
+func findToken(prod *CloudwatchLogs, page *cloudwatchlogs.DescribeLogStreamsOutput) bool {
+	for _, row := range page.LogStreams {
+		if *prod.stream == *row.LogStreamName {
+			prod.token = row.UploadSequenceToken
+			return true
+		}
+	}
+	return false
+}
+
+// Create log group and stream. If an error is returned, PutLogEvents cannot succeed.
+func (prod *CloudwatchLogs) create() (err error) {
+	err = prod.createGroup()
+	if err != nil {
+		return
+	}
+	err = prod.createStream()
+	return
+}
```

Comment:
Fine like this, but you could also do it like this:
```go
func (prod *CloudwatchLogs) create() error {
    if err := prod.createGroup(); err != nil {
        return err
    }
    return prod.createStream()
}
```
I'll leave it up to you which version you like more.